Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1Connections.swift

```diff
@@ -0,0 +1,602 @@
+        }
+
+        var stats: Stats {
+            var stats = Stats()
+            for connectionState in self.connections {
+                if connectionState.isConnecting {
+                    stats.connecting += 1
+                } else if connectionState.isBackingOff {
+                    stats.backingOff += 1
+                } else if connectionState.isLeased {
+                    stats.leased += 1
+                } else if connectionState.isAvailable {
+                    stats.idle += 1
+                }
+            }
+            return stats
+        }
+
+        var isEmpty: Bool {
+            self.connections.isEmpty
+        }
+
+        var canGrow: Bool {
+            if self.overflowIndex < self.connections.endIndex {
+                return self.overflowIndex < self.maximumConcurrentConnections
+            } else {
+                return self.connections.endIndex < self.maximumConcurrentConnections
+            }
+        }
+
+        var startingGeneralPurposeConnections: Int {
+            var connecting = 0
+            for connectionState in self.connections[0..<self.overflowIndex] {
+                if connectionState.isConnecting || connectionState.isBackingOff {
+                    connecting += 1
+                }
+            }
+            return connecting
+        }
+
+        func startingEventLoopConnections(on eventLoop: EventLoop) -> Int {
+            self.connections[self.overflowIndex..<self.connections.endIndex].reduce(into: 0) { count, connection in
+                guard connection.eventLoop === eventLoop else { return }
+                if connection.isConnecting || connection.isBackingOff {
+                    count += 1
+                }
+            }
+        }
+
+        // MARK: - Mutations -
+
+        enum NextConnectionAction<Request> {
+            /// Directly lease the connection again
+            case lease(Connection, Request)
+            /// Start an idle timeout timer and make connection available to new requests
+            case park(Connection.ID)
+            /// Close the connection and remove
+            case close(Connection)
+        }
+
+        // MARK: Connection creation
+
+        mutating func createNewConnection(on eventLoop: EventLoop) -> Connection.ID {
+            let connection = HTTP1ConnectionState(connectionID: self.generator.next(), eventLoop: eventLoop)
+            self.connections.append(connection)
+            self.overflowIndex = self.connections.index(after: self.overflowIndex)
+            return connection.connectionID
+        }
+
+        mutating func createNewOverflowConnection(on eventLoop: EventLoop) -> Connection.ID {
+            let connection = HTTP1ConnectionState(connectionID: self.generator.next(), eventLoop: eventLoop)
+            self.connections.append(connection)
+            return connection.connectionID
+        }
+
+        mutating func newHTTP1ConnectionCreated<RequestType>(
```

Comment:
I _think_ this is called when the connection has been established, "created" doesn't give me that impression.